### PR TITLE
[inductor] adapte windows file path

### PR DESCRIPTION
**Pull request by xuhancn**
This PR is depends on https://github.com/pytorch/pytorch/pull/130132 can be landed successful.
The detailed log: https://github.com/pytorch/pytorch/issues/124245#issuecomment-2211889758

After the file path was adapted for Windows, the first Windows inductor case was run successful.

```python
import torch

def foo(x, y):
    a = torch.sin(x)
    b = torch.cos(x)
    return a + b
opt_foo1 = torch.compile(foo)
print(opt_foo1(torch.randn(10, 10), torch.randn(10, 10)))
```

Result:
image

cc @peterjc123 @mszhanyi @skyline75489 @nbcsm @vladimir-aubrecht @iremyux @Blackhex @cristianPanaite @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/cpp_builder.py [hunk 1]
```diff
@@ -244,6 +244,12 @@ def run_command_line(cmd_line, cwd=None):
     return status
 
 
+def adapte_file_path(orig_path: str) -> str:
```
Comment (jgong5) [inline torch/_inductor/cpp_builder.py:247]: `normalize_path_separator` sounds more specific.

File: torch/_inductor/cpp_builder.py [hunk 2]
```diff
@@ -244,6 +244,12 @@ def run_command_line(cmd_line, cwd=None):
...
+    if _IS_WINDOWS:
+        return orig_path.replace("\\", "/")
```
Comment (jgong5) [inline torch/_inductor/cpp_builder.py:249]:
```suggestion
        return orig_path.replace(os.sep, "/")
```

**Review by jgong5 — DISMISSED** [review]

File: torch/_inductor/cpp_builder.py (hunk 1 above)
Comment (xuhancn) [reply torch/_inductor/cpp_builder.py]: Done.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Review by desertfire — APPROVED** [review]
Test files need to be updated because of "Unexpected success"

**Comment by xuhancn** [comment]
> @pytorchbot revert -m "broke functorch\test_eager_transforms.py on windows https://github.com/pytorch/pytorch/actions/runs/9958208725/job/27530132704 https://hud.pytorch.org/pytorch/pytorch/commit/69e99172450e40536bf2e6c110183d34a0e283e2. Test failure on PR is real, possibly force merged to get around lint error?" -c ignoredsignal

Lint error is unchanged file, it is not bring from me. @clee2000
